### ISSUE TITLE
Objective-C message was sent to a deallocated 'GDTCCTUploadOperation' object (zombie)

### ISSUE THREAD

**Issue by Pixette** [firebase/firebase-ios-sdk#8472]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.5
  * Firebase SDK version: 8.3.0

Firebase/AnalyticsWithoutAdIdSupport (8.3.0):
  - Firebase/CoreOnly (8.3.0):
  - Firebase/Crashlytics (8.3.0):
  - FirebaseAnalytics/WithoutAdIdSupport (8.3.0):
  - FirebaseCore (8.3.0):
  - FirebaseCoreDiagnostics (8.3.0):
  - FirebaseCrashlytics (8.3.0):
  - FirebaseInstallations (8.3.0):
  - GoogleAppMeasurement/WithoutAdIdSupport (8.3.0):
  - GoogleDataTransport (9.0.1):
  - GoogleUtilities/AppDelegateSwizzler (7.4.3):
  - GoogleUtilities/Environment (7.4.3):
  - GoogleUtilities/Logger (7.4.3):
  - GoogleUtilities/MethodSwizzler (7.4.3):
  - GoogleUtilities/Network (7.4.3):
 

  * Installation method: CocoaPods
  * Firebase Component: Crashlytics + AnalyticsWithoutAdIdSupport

#### Steps to reproduce:

My app creates a slideshow of pictures and movies collected through WebDAV from a NAS Server (e.g. Synology). 
I use only Firebase/Crashlytics + Firebase/AnalyticsWithoutAdIdSupport pods.

On older **devices (iOS 9.3.5)**, following crash occurs:
- The app plays automatically a movie/picture each 5 seconds (slideshow)
- I do not fire Analytics events during this "automatic playing of files" (I only fire analytics upload during user interaction)
- (However the SDK certainly must upload data on a regular basis automatically)
- Suddenly after > 30min to 1-2 hours (random delay), the app crash
- Using Xcode Zombies profile, I could get following error during crash : "An Objective-C message was sent to a deallocated 'GDTCCTUploadOperation' object (zombie) at address: 0x1e2e3550. "

**This error doesn't seems to happen for newer devices on higher iOS version. I presume that because on Crashlytics, I receive some unknown heap corruption errors (see below) which I think are related to those crashes and those are only reported on iOS 9.3.5 devices**

#### Relevant Code:
XCODE:

![image](https://user-images.githubusercontent.com/83535288/127748416-8797e65d-17c5-474e-bc56-e3fc659fc263.png)

![image](https://user-images.githubusercontent.com/83535288/127748426-8a174f44-de76-4233-a95d-4356c3794c6c.png)

I presume that the following "Heap Corruption" error I gathered on Crashlytics and which only occurs on iOS 9.3.5 devices is probably the same consequence of above mentioned problem:

![image](https://user-images.githubusercontent.com/83535288/127748455-0d369f24-782e-4cd9-9d42-ba5d3bfa7d51.png)

DEVICE CRASH LOG:
[crash140821.log](https://github.com/firebase/firebase-ios-sdk/files/6986065/crash140821.log)

<img width="2002" alt="Screenshot 2021-08-14 at 08 50 51" src="https://user-images.githubusercontent.com/83535288/129438497-c4e95aa0-8881-49c7-9edc-6491fbcfd6e4.png">

**Comment by paulb777** [comment]
@maksymmalyhin Any ideas or suggestions here?

**Comment by Pixette** [comment]
I could add a device crash log and a screenshot of the crash when xcode was attached to the device.

**Comment by maksymmalyhin** [comment]
@Pixette The issue might be related to #8321 though it looks pretty different. #8321 happens only when Firebase Performance is installed with instrumentation enabled. Could you confirm if Firebase Performance is installed in your app?

**Comment by Pixette** [comment]
@maksymmalyhin  No only following pods were used : Firebase/**Crashlytics** + Firebase/**AnalyticsWithoutAdIdSupport**

I tried as best as I can to disable all uses of Firebase on iOS 9.3 to avoid the crash. Unfortunately, I could narrow it down that having only the "import firebase" on top of my file was enough to crash. I tried following without success:
- Set FIREBASE_ANALYTICS_COLLECTION_ENABLED to false in my info.plist (same result as with setting it in the code with: Analytics.setAnalyticsCollectionEnabled(false))
- Execute Firebase.configure() only on iOS 10+
- Disabling all code executing Firebase or Analytics events on iOS 9.3

As I don't know how technically I could import Firebase only for end-users with iOS 10+, I need to disable it completely for the moment.
